addons/mail/models/mail_message.py

```diff
@@ -285,10 +285,7 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         for message in message_tree.values():
             if message.author_id:
                 partners |= message.author_id
-            if message.subtype_id and message.partner_ids:  # take notified people of message with a subtype
-                partners |= message.partner_ids
-            elif not message.subtype_id and message.partner_ids:  # take specified people of message without a subtype (log)
-                partners |= message.partner_ids
+            partners |= message.partner_ids
```

Comment:
Not cool 🙂 

The original `if-elif` block was kinda weird, yeah, but that condition still took `message.partner_ids` into account
I don't really think that removing the whole condition is a good idea, and that's why:

With 
```python
x = message.subtype_id
y = message.partner_ids
```
and using `foo()` as an alias for `partners |= message.partner_ids`, this roughly translates to:
```python
if x and y:
    foo()
elif not x and y:
    foo()
```
Furthermore:
```python
if (x and y) or (not x and y):
    foo()
```
At this point, this is safe to throw `x` expression part out of the condition:
```python
if y:
    foo()
```
Just translate it back to original variables, we'll get
```python
if message.partner_ids:
    partners |= message.partner_ids
```
...but the condition is still there. Keep it pls. 🙂